osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -198,37 +204,45 @@ protected override void LoadComplete()
         }
 
         /// <summary>
-        /// Applies a new <see cref="HitObject"/> to be represented by this <see cref="DrawableHitObject"/>.
+        /// Applies a hit object to be represented by this <see cref="DrawableHitObject"/>.
         /// </summary>
-        /// <param name="hitObject">The <see cref="HitObject"/> to apply.</param>
-        /// <param name="lifetimeEntry">The <see cref="HitObjectLifetimeEntry"/> controlling the lifetime of <paramref name="hitObject"/>.</param>
+        /// <remarks>This overload is semi-deprecated. Use either <see cref="Apply(osu.Game.Rulesets.Objects.HitObject)"/> or <see cref="Apply(osu.Game.Rulesets.Objects.HitObjectLifetimeEntry)"/>.</remarks>
         public void Apply([NotNull] HitObject hitObject, [CanBeNull] HitObjectLifetimeEntry lifetimeEntry)
         {
-            free();
+            if (lifetimeEntry != null && lifetimeEntry.HitObject != hitObject)
+                throw new InvalidOperationException($"{nameof(HitObjectLifetimeEntry)} has different {nameof(HitObject)} from the specified one.");
 
-            HitObject = hitObject ?? throw new InvalidOperationException($"Cannot apply a null {nameof(HitObject)}.");
+            if (lifetimeEntry != null)
+                Apply(lifetimeEntry);
+            else
+                Apply(hitObject);
+        }
 
-            this.lifetimeEntry = lifetimeEntry;
+        /// <summary>
+        /// Applies a new <see cref="HitObject"/> to be represented by this <see cref="DrawableHitObject"/>.
+        /// A new <see cref="HitObjectLifetimeEntry"/> is automatically created and applied to this <see cref="DrawableHitObject"/>.
+        /// </summary>
+        public void Apply([NotNull] HitObject hitObject)
+        {
+            if (hitObject == null)
+                throw new ArgumentNullException($"Cannot apply a null {nameof(HitObject)}.");
 
-            if (lifetimeEntry != null)
-            {
-                // Transfer lifetime from the entry.
-                LifetimeStart = lifetimeEntry.LifetimeStart;
-                LifetimeEnd = lifetimeEntry.LifetimeEnd;
+            Apply(new UnmanagedHitObjectEntry(hitObject));
+        }
 
-                // Copy any existing result from the entry (required for rewind / judgement revert).
-                Result = lifetimeEntry.Result;
-            }
-            else
-                LifetimeStart = HitObject.StartTime - InitialLifetimeOffset;
```

Comment:
This behaviour seems to have been lost somewhere along the way (if no entry is given, lifetime start is automatically calculated in this manner; I see this being removed but not added back anywhere else) - I believe we have consumers relying on it, so it needs to remain.

Reply:
I thought it was fine because `LifetimeStart` is still set in `HitObjectLifetimeEntry` but actually it uses `HitObjectLifetimeEntry`'s `InitialLifetimeOffset` which is different from `DrawableHitObject`'s `InitialLifetimeOffset` so it has to be fixed to not accidentally change the behavior.